vault-github/run.sh

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env bash
+
+VAULT_ADDR=${VAULT_ADDR}
+VERSION=${VERSION}
+AUTH_PATH=${AUTH_PATH}
+GITHUB_EXTERNAL_AUTH_ID=${GITHUB_EXTERNAL_AUTH_ID}
+
+# Ensure curl is installed
+if ! command -v curl &>/dev/null; then
+    printf "curl is not installed. Please install curl in your image.\n"
+    exit 1
+fi
+
+# Fetch latest version of Vault if VERSION is 'latest'
+if [ "${VERSION}" = "latest" ]; then
+    LATEST_VERSION=$(curl -s https://releases.hashicorp.com/vault/ | grep -oP 'vault/\K[0-9]+\.[0-9]+\.[0-9]+' | sort -V | tail -n 1)
+    if [ -z "$LATEST_VERSION" ]; then
+        printf "Failed to determine the latest Vault version.\n"
+        exit 1
+    fi
+    VERSION=$LATEST_VERSION
+fi
+
+# Check if vault is installed and has the correct version
+installation_needed=1
+if command -v vault &>/dev/null; then
+    CURRENT_VERSION=$(vault version | grep -oE '[0-9]+\.[0-9]+\.[0-9]+')
+    if [ "$CURRENT_VERSION" = "$VERSION" ]; then
+        printf "Vault version $CURRENT_VERSION is already installed and up-to-date.\n\n"
```

Comment:
Variables with `printf` is not recommended:

```suggestion
        printf "Vault version %s is already installed and up-to-date.\n\n" "$CURRENT_VERSION"
```